controls/V-72037.rb

```diff
@@ -79,17 +80,37 @@
     #For each user, build and execute a find command that identifies initialization files
     #in a user's home directory.
     u.each do |user|
-      dotfiles = dotfiles + command("find #{user.home} -xdev -maxdepth 2 -name '.*' -type f").stdout.split("\n")
+      dotfiles = dotfiles + command("find #{user.home} -xdev -maxdepth 2 ( -name '.*' ! -name '.bash_history' ) -type f").stdout.split("\n")
     end
     ww_files = Set[]
-    ww_files = command('find / -perm -002 -type f -exec ls {} \;').stdout.lines
+    ww_files = command('find / -xdev -perm -002 -type f -exec ls {} \;').stdout.lines
+
+    #To reduce the number of commands ran, we use a pattern file in the grep command below
+    #So we don't have too long of a grep command, we chunk the list of ww_files
+    #into strings not longer than PATTERN_FILE_MAX_LENGTH
+    PATTERN_FILE_MAX_LENGTH=100000
```

Comment:
I would base this off of the result of `getconf ARG_MAX` and have a reasonable limit (this seems fine) if that doesn't return anything.

Reply:
Since we are passing the pattern file as one argument, it looks like our limit will be MAX_ARG_STRLEN, which `getconf` can't get. It's hard-coded in /usr/include/linux/binfmts.h as `#define MAX_ARG_STRLEN (PAGE_SIZE * 32)`. Added a commit to address this.